libr/core/cmd_print.c

```diff
@@ -4671,7 +4671,7 @@ static int cmd_print(void *data, const char *input) {
 	ut32 tbs = core->blocksize;
 	ut64 n, off, from, to, at, ate, piece;
 	ut64 tmpseek = UT64_MAX;
-	const int addrbytes = core->io->addrbytes;
+	const size_t addrbytes = core->io->addrbytes;
```

Comment:
`addrbytes` is `int` in `RIO`, why this change?

Reply:
We have a possible overflow bellow, if the type is int.

```
size_t ... = int * int;
                 ^
                 | -- possible overflow
```

Reply:
I see, but I'm not sure that's the proper way to fix it (since you store the result in a `int` anyway). Maybe just check for overflows before doing the multiplication. It is probably wrong anyway if the number is so big.

Reply:
What about a conversion like (size_t)int * int? Or change addrbytes to some ut if it doesn't use -1 to indicate error.

Reply:
that wont solve the problem. we have the UT64_ADD_OVFCHK macros that detect the overflow before it happens. but we have it for + not for *.  imho addrbytes shouldnt be larger than 4 or 8, so we probably dont need to be that paranoic on handling overflows in this operation

Reply:
swap back to the old type